src/database/services/vacation-request-api-service.ts

```diff
@@ -43,35 +44,50 @@ class VacationRequestService {
         TableName: TABLE_NAME,
         Key: {
           id: id
-        },
+        }
       })
       .promise();
 
     return result.Item as VacationRequestModel;
-  }
+  };
 
   /**
    * Lists all vacation requests
    *
    * @returns list of vacation requests
    */
-  public listVacationRequests = async (): Promise<VacationRequestModel[]> => {
+  public listVacationRequests = async (userId?: string): Promise<VacationRequestModel[]> => {
+    if (userId) {
+      const filteredResult = await this.docClient
+        .scan({
+          TableName: TABLE_NAME,
+          FilterExpression: "user = :userId",
```

Comment:
Is it possible for the filterExpression to be undefined/ null, that would allow the userId condition to be handled in a single line rather than needing to have filteredResult or result